satpy/readers/netcdf_utils.py

```diff
@@ -258,13 +258,17 @@ def collect_cache_vars(self, cache_var_size):
         cache_vars = self._collect_cache_var_names(cache_var_size)
         for var_name in cache_vars:
             v = self.file_content[var_name]
+            try:
+                attrs = v.attrs
+            except AttributeError:
+                attrs = v.__dict__
```

Comment:
I'm not very familiar with this, but it feels like it deserves a test...

Reply:
I agree, but I have no idea how to test. We don't access any real files in the tests.

Reply:
Maybe we then should add one test with a real file?

Reply:
I have an idea, lets see if I can complete it today.

Reply:
https://github.com/pytroll/satpy/pull/2753/commits/f1bc699f8a80bad86607901aa5954b8ebb8f969c should cover the addition. Fails with the original and passes with the current implementation for `h5netcdf` backend (remote reading).